Python/bltinmodule.c

```diff
@@ -2479,7 +2479,15 @@ builtin_sum_impl(PyObject *module, PyObject *iterable, PyObject *start)
                 return PyLong_FromLong(i_result);
             }
             if (PyLong_CheckExact(item) || PyBool_Check(item)) {
-                long b = PyLong_AsLongAndOverflow(item, &overflow);
+                long b;
+                overflow = 0;
+                /* Single digits are common, fast, and cannot overflow on unpacking. */
+                switch (Py_SIZE(item)) {
+                    case -1: b = -(sdigit) ((PyLongObject*)item)->ob_digit[0]; break;
```

Comment:
Should the *sdigit* cast be in case 1 as well?  Or can it be dropped in both places?

Reply:
I can't be removed from the negative case because `digit` is unsigned and thus can't be negated without the cast.
Since negating is the reason for the cast, I don't think it's needed for the positive case.